### ISSUE TITLE
Customizable shape for the Line Arrowheads

### ISSUE THREAD

**Issue by zweek** [motion-canvas/motion-canvas#380]
**Description**
Right now there seems to be no way to customize the appearance of the arrowheads (beyond the `arrowSize` property), so they are always drawn as the same right triangle shape

**Proposed solution**
To have maybe 2 extra properties - one for 'sharpness' that controls the angle at the tip and one radius property for rounding the corners of the triangle. That way you could tweak those properties to get an arrowhead that could look like the one in the screenshot below

**Additional context**
![image](https://user-images.githubusercontent.com/70382336/219942849-aa5b2ed3-ab90-479b-95ce-fd26098dec6d.png)

**Comment by strau0106** [comment]
I'd propose that you can pass another shape object as the Arrow Head. Because otherwise we would reimplement features that already exist.

**Comment by zweek** [comment]
Yeah I like that idea a lot more, actually. That also opens up a lot more possibilities for what you can do with Lines if you can pass in whatever shape you want

**Comment by strau0106** [comment]
Actually not a Shape but a Node in general, considering that right now triangles and other non 4-edged shapes are made with lines

**Comment by aarthificial** [comment]
`Line` extends `Shape` but I agree that we should accept any `Node`.

I propose extending the `startArrow` `endArrow` properties to `Node | bool`.
The `Node.draw` method can be used to reuse the drawing logic of a Node. This method assumes that the canvas' matrix has been already transformed to the node's local space (`[0, 0]` being the center of the node). This way we won't need to reparent anything.

**Comment by strau0106** [comment]
Sounds good! 

I will be claiming this issue if that‘s okay for everyone!

**Comment by strau0106** [comment]
The problem I'm seeing, is that the draw function on a Node is protected. Thus I can't call it from the drawShape function in Line for the respective ArrowHeads
